frontend/src/components/WorksheetExpandedList.js

```diff
@@ -38,13 +55,56 @@ const WorksheetExpandedList = ({
 }) => {
   const { user } = useUser();
   const [isList, setView] = useState(true);
+  // State that controls sortby select
+  const [select_sortby, setSelectSortby] = useState(sortbyOptions[0]);
+  // State that determines sort order
+  const [sort_order, setSortOrder] = useState('asc');
+
+  const handleSortOrder = () => {
+    if (sort_order === 'asc') setSortOrder('desc');
+    else setSortOrder('asc');
+  };
+
+  const WorksheetData = useMemo(() => {
+    let sortParams = select_sortby.value;
+    let ordering = null;
+    if (sortParams === 'course_code') {
+      ordering = { course_code: sort_order };
+    } else if (sortParams === 'course_title') ordering = { title: sort_order };
+    else if (sortParams === 'course_number') ordering = { number: sort_order };
+    else if (sortParams === 'rating') ordering = { average_rating: sort_order };
+    else if (sortParams === 'workload')
+      ordering = { average_workload: sort_order };
+    else if (sortParams === 'professor')
+      ordering = { average_professor: `${sort_order}_nulls_last` };
+    else if (sortParams === 'gut')
+      ordering = { average_gut_rating: `${sort_order}_nulls_last` };
+    else console.error('unknown sort order - ', sortParams);
+
+    // Apply sorting order.
+    // Force anything that is null to the bottom.
+    // In case of ties, we fallback to the course code.
+    const key = Object.keys(ordering)[0];
+    const order_asc = ordering[key].startsWith('asc');
+    const listings = orderBy(
+      courses,
+      [
+        (listing) => !!listing[key],
+        (listing) => listing[key],
+        (listing) => listing.course_code,
+      ],
+      ['desc', order_asc ? 'asc' : 'desc', 'asc']
+    );
+    return listings;
+  }, [courses, select_sortby, sort_order]);
```

Comment:
Is there a way to refactor this logic out - don't want to duplicate logic between the search and here